### PR TITLE
feat(utils): Data schema validation

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Introduce schema validation helper and make it validate Pull Request coercion result.
It ensures schema error of the same kind is reported just once.

## Context

- Ref: #13992

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Example error:

image

**Review by rarkins — APPROVED** [review]

File: lib/util/schema.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+export function checkSchema<T extends z.ZodSchema>(
+  schema: T,
+  input: unknown
+): input is z.infer<T> {
+  const res = schema.safeParse(input);
+  const { success } = res;
+  if (!success) {
+    collectError(schema, res.error);
```
Comment (viceice) [inline lib/util/schema.ts:54]: why do we defer error reporting? isn't it hard to find right code later then?

Comment (zharinov) [reply lib/util/schema.ts:54]: Can be found by stacktrace. For now, it's more like additional stats reporting rather than runtime errors.

Comment (zharinov) [reply lib/util/schema.ts:54]: Oh yes, we do, `collectError` handles it

**Pull request merged**